### DISCUSSION TITLE
Can I use CustomCSSforFx, Menu Icon Plus CSS or other GUI modifying CSS through StyloaiX? How?

### DISCUSSION THREAD

**Discussion by mzso** [xiaoxiaoflood/firefox-scripts#240] [General Support]
Hi!

Since modifying stuff through userchrome/usercontent is very cumbersome and annoying I was wondering if I can use these CSS packages through StyloaiX, and particularly view their changes in a running instance without having to restart the browser every time I change something?

If it even works, can I do more than import the packages' "userchrome.css" file in a StyloaiX style? Which wouldn't allow me to edit the styles within the browser? (As far as I understand)

**Comment by mzso** [comment]
Well, I tried it. The import paths suck though. I had to use ../CustomCSSforFx/<etc>
But this is pretty vulnerable. If the location of StyloaiX css changes everything would break. How do I access the chrome folder or the profile folder directly?

**Comment by 117649** [comment]
I've done something similar recently.

I use this:
```css
/** Import Lepton stylesheet (Chrome) *****************************************/
@import url("css/leptonChrome.css");
@import url("css/leptonContent.css");

/** Add your custom styles below **********************************************/
```
 to load [Lepton](https://github.com/black7375/Firefox-UI-Fix).

But I just tossed everything of it in to the \UserStyles folder of StyloaiX.

**Reply by mzso** [reply]
Well, I guess it only works relative to the css file you're using. So if you move it everything will break.
That's why it would be nice if something like ff-profile-folder/chrome/whatever.css or chrome-folder/whatever.css would work.
I figured a "chrome://" url might do this, but I could find it how, if it's possible.

**Reply by 117649** [reply]
How about "file://"?

**Reply by mzso** [reply]
And use absolute system paths?
The has whole other issues with moving profiles, or copying css customizations.

**Reply by 117649** [reply]
resource://userchromejs/

**Reply by mzso** [reply]
Thanks. That points to \<userprofile>\/chrome?

**Reply by mzso** [reply]
> resource://userchromejs/

I replaced all instances of "../" with it. Seems like it works fine, and exactly how I wanted it.

(BTW is this resource defined by Firefox itself or the userchrome js loader?)

**Reply by 117649** [reply]
> > resource://userchromejs/
> 
> I replaced all instances of "../" with it. Seems like it works fine, and exactly how I wanted it.
> 
> (BTW is this resource defined by Firefox itself or the userchrome js loader?)

https://github.com/xiaoxiaoflood/firefox-scripts/blob/b013243f1916576166a02d816651c2cc6416f63e/chrome/utils/chrome.manifest#L1-L2

**Reply by TFWol** [reply]
Good reference where the scheme is talked about by xiaoxiaoflood:
https://github.com/xiaoxiaoflood/firefox-scripts/issues/189#issuecomment-1278247509

> StyloaiX's styles path is like resource://userchromejs/UserStyles/autohide_sidebar.css. The root resource://userchromejs/ is the chrome folder from running profile.
> 
> If you need to import from outside this context, you must use full path like file:///home/user/.mozilla/firefox/aeskq86n.default-ahmed/chrome/firefox-csshacks/chrome/autohide_sidebar.css.
> 
> If unsure of full path, just drag the file from your file manager to Firefox and copy the value in addressbar.
